### DISCUSSION TITLE
What is the purpose of AddCoords1d?

### DISCUSSION THREAD

**Discussion by ksachdeva** [timeseriesAI/tsai#251] [Q&A] [answered]
Hi,

Thanks for this great work.

Unfortunately, I have not been able to understand the purpose of this layer i.e. `AddCoords1d`.

I have experimented with it and understand what it is doing but not sure why it is doing so or where is this extra dimension/channel is being used?

Please guide.

Regards & thanks
Kapil

**Comment by vrodriguezf** [comment, marked as answer]
I think it has to do with the research in CoordConv layers that Uber popularized a few years ago:
https://arxiv.org/abs/1807.03247

**Comment by ksachdeva** [comment]
Thanks @vrodriguezf 

Much appreciated the quick answer and the pointer. 

Reading the paper and indeed it seems to be the implementation of this paper.

Regards
Kapil
